xbmc/platform/posix/filesystem/PosixFile.cpp

```diff
@@ -22,6 +22,11 @@
 #include <fcntl.h>
 #include <sys/ioctl.h>
 #include <sys/stat.h>
+
+#if defined(STATX_TYPE) // use statx if available to get file birth date
+#define HAS_STATX 1
```

Comment:
can this check be done in cmake? similar to what is done here https://github.com/xbmc/xbmc/blob/master/cmake/scripts/linux/ArchSetup.cmake#L86-L123

Reply:
it sure can, but is that better than this check?

Reply:
might be nice to have the message if a feature isn't found, rather than blindly building with or without a feature

Reply:
If I do the check in cmake, will that avoid the android clash, where it seems that the statx flags are defined but not the function itself?

Reply:
Sure, you could just include the check in `linux/ArchSetup.cmake` and leave it out of `android/ArchSetup.cmake`

Reply:
the check should probably be run for all POSIX platforms. It's possible android (and others) will support this in the future

Reply:
I think I can do that now I have figured out why the X11 build failed yesterday.  The required flags can be defined in `stat.h` but if the C library is too old, there is no matching function call (glibc needs to be 2.28 or greater).  I believe Android _does_ support it in Bionic or with glibc >= 2.28 (https://android.googlesource.com/platform/bionic.git/+/master/tests/sys_stat_test.cpp)